### DISCUSSION TITLE
How can I make `username` field required?

### DISCUSSION THREAD

**Discussion by babakfp** [pocketbase/pocketbase#1743] [Q&A] [answered]
![Screenshot 2023-02-01 130030](https://user-images.githubusercontent.com/44144724/216165149-8c7fd787-b525-4ecb-be78-511dbafbc058.png)

**Comment by ganigeorgiev** [comment, marked as answer]
`username` is optional during create to preserve the backward compatibility and to prevent getting in the way to those who don't use it. As the field description mention, even if the user doesn't submit the field during create, it will be autogenerated (eg. the name of the collection + some random numbers).

You can "enforce" the username field to be required during create either client-side and/or define a "Create API" rule like `@request.data.username != ""` (_I personally will go with only the client-side option because as mentioned even if the user doesn't submit a value it will be autogenerated_).

**Reply by babakfp** [reply]
![Screenshot 2023-02-01 134450](https://user-images.githubusercontent.com/44144724/216170326-2633a2e7-f675-4858-af07-346c44839761.png)

So, I did that and it throws this error (instead of something like when you send an empty string for the password field):

```
{
  url: 'http://127.0.0.1:8090/api/collections/users/records',
  status: 400,
  data: { code: 400, message: 'Failed to create record.', data: {} },
  isAbort: false,
  originalError: null
}
```

**Reply by ganigeorgiev** [reply]
Sorry, I forgot to mention it. This is working as intended. The rules don't return individual field/condition errors.

**Reply by babakfp** [reply]
Alright
Thanks
